File: geocoder-openstreetmap/src/main/java/com/prolificinteractive/geocoder/api/openstreetmaps/OSMAddressDetail.java
package com.prolificinteractive.geocoder.api.openstreetmaps;

import android.support.annotation.Nullable;
import com.google.gson.annotations.SerializedName;

class OSMAddressDetail {

  @SerializedName("country")
  @Nullable public String country;

  @SerializedName("country_code")
  @Nullable public String countryCode;

  @SerializedName("city")
  @Nullable public String city;

  @SerializedName("county")
  @Nullable public String county;

  @SerializedName("postcode")
  @Nullable public String postcode;

  @SerializedName("neighbourhood")
  @Nullable public String neighbourhood;

  @SerializedName("suburb")
  @Nullable public String suburb;

  @SerializedName("state")
  @Nullable public String state;
}

File: geocoder-openstreetmap/src/main/java/com/prolificinteractive/geocoder/api/openstreetmaps/OSMAddressDetail.kt
package com.prolificinteractive.geocoder.api.openstreetmaps

import com.google.gson.annotations.SerializedName

data class OSMAddressDetail(

  @SerializedName("country")
  var country: String? = null,

  @SerializedName("country_code")
  var countryCode: String? = null,

  @SerializedName("city")
  var city: String? = null,

  @SerializedName("county")
  var county: String? = null,

  @SerializedName("postcode")
  var postcode: String? = null,

  @SerializedName("neighbourhood")
  var neighbourhood: String? = null,

  @SerializedName("suburb")
  var suburb: String? = null,

  @SerializedName("state")
  var state: String? = null
)